EXPLAINER_BLUETOOTH.md

```diff
@@ -0,0 +1,227 @@
+   port.)
+
+## What does this proposal add?
+
+This proposal adds the ability for the browser to:
+
+1. Enumerate bonded Bluetooth devices that expose a serial interface using the
+   standard Serial Port service.
+2. Communicate with the serial interface even if the operating system has not
+   created a device node for that port.
+3. Communicate with a non-Serial Port service that exposes a serial interface.
+   See [Non-standard Service Class ID’s](#non-standard-service-class-ids) below.
+
+## Why not enhance Web Bluetooth?
+
+The intent is to avoid the tendency to backfill Bluetooth Classic support into
+[Web Bluetooth](https://webbluetoothcg.github.io/web-bluetooth/). If support for
+one profile were to land in Web Bluetooth, this might entice developers to
+request the addition of other classic profiles to Web Bluetooth. Support for
+Bluetooth serial interfaces is seen as a special case as It has come numerous
+times in the form of developer requests needing basic serial port support with
+no simple alternative in BLE.
+
+In short, this feature is a natural extension to the Web Serial API, and will
+minimize requests for other classic profiles.
+
+## Support for unmapped serial ports
+
+When bonding to a Bluetooth device that exposes a serial port using the standard
+Serial Port service, some operating systems will automatically create device
+nodes representing the serial interface. Windows creates a pair of COM ports,
+and macOS a pair of filesystem nodes in /dev (i.e. `/dev/cu.*` and
+`/dev/tty.*`).
+
+This proposal will allow support for Bluetooth services that implement the
+Serial Port Profile even if the operating system does not create device nodes
+for Bluetooth serial ports.
+
+## Non-standard Service Class ID’s
+
+Devices supporting a Serial Port Profile
+([SPP](https://www.bluetooth.com/specifications/specs/serial-port-profile-1-2/))
+often offer this via the standard Serial Port service (UUID: `0x1101`).
+Operating systems may create devices or ports mapped to this Bluetooth service
+during the bonding process. On macOS and Windows, two device nodes are created
+for each bonded device that exposes a serial port using the standard Serial Port
+service.
+
+This proposal recommends supporting **any** service that implements the Serial
+Port Profile, regardless of whether the service is a standard Serial Port
+service or is automatically mapped by the operating system. By default
+`Serial.requestPort()` will only include Bluetooth serial ports exposed by
+devices with a standard Serial Port service. A new service UUID filter option is
+added to `Serial.requestPort()` to enable requesting serial ports for other
+services.
+
+## Potential specification changes
+
+### Port Selection
+
+[`Serial.requestPort()`](https://developer.mozilla.org/en-US/docs/Web/API/Serial/requestPort),
+with no filters parameter, will pass through:
+
+1. Non-Bluetooth serial ports (wired, virtual, etc.)
+2. Mapped Bluetooth serial ports.
```

Comment:
It's not clear at this point what "mapped" means. I think you mean bonded Bluetooth devices where the operating system has automatically created device nodes for the Serial Port service.

Reply:
I discuss this tangentially in "Support for unmapped serial ports" above (currently line 66). I'm using "mapped" to describe the creation of one (or more) serial device nodes existing for a single Bluetooth serial port. They can be automatically created (which Windows and macOS do), or manually (like using `rfcomm` on Linux). Do you think "map" is a good term for this, or is there a better one?

Reply:
I think I withdraw the question because I just didn't read the section heading above carefully enough.